### PR TITLE
Support report usage api

### PR DESCRIPTION
**Pull request by AlexeiVainshtein**

### CODE REVIEW

File: artifactory/reportusage/reportusage.go [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+	bodyContent, err := json.Marshal(params)
+	return bodyContent, err
+}
+
+type reportUsageParams struct {
+	ProductId string          `json:"productId"`
+	Features  []featureParams `json:"features,omitempty"`
+}
+
+type featureParams struct {
```
Comment (eyalbe4) [inline artifactory/reportusage/reportusage.go]: I suggest renaming the struct name to *features*.

File: artifactory/reportusage/reportusage.go [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+	if resp.StatusCode != http.StatusOK {
+		return errors.New("Artifactory response: " + resp.Status + "\n" + clientutils.IndentJson(body))
+	}
+
+	log.Debug("Artifactory response:", resp.Status)
+	return nil
+}
+
+// Returns an error if the Artifactory version is not compatible
+func isCompatible(artifactoryVersion string) bool {
```
Comment (eyalbe4) [inline artifactory/reportusage/reportusage.go]: I suggest renaming the function name to *isSupportedVersion*.
If we change the name, let's also rename the test for this function accordingly.

Comment (AlexeiVainshtein) [reply artifactory/reportusage/reportusage.go]: Per our discussion, changed it to isVersionCompatible.

File: artifactory/reportusage/reportusage.go [hunk 3]
```diff
@@ -0,0 +1,80 @@
+package reportusage
```
[the remark this answers is not in the document]
Comment (AlexeiVainshtein) [reply artifactory/reportusage/reportusage.go]: Done

File: artifactory/reportusage/reportusage.go [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+	"github.com/jfrog/jfrog-client-go/artifactory"
+	"github.com/jfrog/jfrog-client-go/artifactory/services/utils"
+	clientutils "github.com/jfrog/jfrog-client-go/utils"
+	"github.com/jfrog/jfrog-client-go/utils/log"
+	"github.com/jfrog/jfrog-client-go/utils/version"
+	"github.com/pkg/errors"
+	"net/http"
+)
+
+const SystemUsageApi = "6.9.0"
```
[the remark this answers is not in the document]
Comment (AlexeiVainshtein) [reply artifactory/reportusage/reportusage.go]: Done

File: artifactory/reportusage/reportusage.go [hunk 5]
```diff
@@ -0,0 +1,80 @@
...
+
+func SendReportUsage(productId, commandName string, serviceManager *artifactory.ArtifactoryServicesManager) error {
+	config := serviceManager.GetConfig()
+	if config == nil {
+		return errors.New("Expected full config, but no configuration were configured.")
```
[the remark this answers is not in the document]
Comment (AlexeiVainshtein) [reply artifactory/reportusage/reportusage.go]: Done

File: artifactory/reportusage/reportusage.go [hunk 6]
```diff
@@ -0,0 +1,80 @@
...
+	}
+	rtDetails := config.GetArtDetails()
+	if rtDetails == nil {
+		return errors.New("Expected Artifactory details but no details were configured")
```
[the remark this answers is not in the document]
Comment (AlexeiVainshtein) [reply artifactory/reportusage/reportusage.go]: Done

File: artifactory/reportusage/reportusage.go [hunk 7]
```diff
@@ -0,0 +1,80 @@
...
+		return err
+	}
+	clientDetails := rtDetails.CreateHttpClientDetails()
+	// Check Artifactory version
+	artifactoryVersion, err := rtDetails.GetVersion()
+	if err != nil {
+		return err
+	}
+	if !isCompatible(artifactoryVersion) {
+		return fmt.Errorf("Expected Artifactory version %s or above, got %s", SystemUsageApi, artifactoryVersion)
```
[the remark this answers is not in the document]
Comment (AlexeiVainshtein) [reply artifactory/reportusage/reportusage.go]: I've changed it

File: artifactory/reportusage/reportusage.go [hunk 8]
```diff
@@ -0,0 +1,80 @@
+package reportusage
+
```
[the remark this answers is not in the document]
Comment (AlexeiVainshtein) [reply artifactory/reportusage/reportusage.go]: Added the check error

File: artifactory/reportusage/reportusage.go [hunk 9]
```diff
@@ -0,0 +1,80 @@
...
+	}
+
+	log.Debug("Artifactory response:", resp.Status)
+	return nil
+}
+
+// Returns an error if the Artifactory version is not compatible
+func isCompatible(artifactoryVersion string) bool {
+	// API exists from Artifactory version 6.9.0 and above:
+	if version.Compare(artifactoryVersion, SystemUsageApi) < 0 && artifactoryVersion != "development" {
```
[the remark this answers is not in the document]
Comment (AlexeiVainshtein) [reply artifactory/reportusage/reportusage.go]: Done

File: artifactory/reportusage/reportusage.go (hunk 1 above)
Comment (AlexeiVainshtein) [reply artifactory/reportusage/reportusage.go]: Changed it to feature

**Pull request merged**